lib/ts/nextjs.ts

```diff
@@ -231,7 +252,20 @@ export default class NextJS {
                 return nextResponse as NextResponse;
             }
 
-            let userResponse = await handler(undefined, session);
+            let userResponse: NextResponse;
+
+            try {
+                userResponse = await handler(undefined, session);
+            } catch (err) {
+                await customErrorHandler()(err, baseRequest, baseResponse, (errorHandlerError: Error) => {
+                    if (errorHandlerError) throw errorHandlerError;
+                });
+
+                userResponse = new Response(baseResponse.body, {
+                    status: baseResponse.statusCode,
+                    headers: baseResponse.headers,
+                }) as NextResponse;
```

Comment:
we set this here, and below too we transfer the cookies and headers once again.. how does this work?

Reply:
In some cases, headers will be set twice but that is fine. I have tested that headers only appear once in the response.